src/Vegetation/PlantHydraulics.jl

```diff
@@ -326,72 +423,83 @@ end
 
 """
     make_rhs(model::VegetationModel)
+
 A function which creates the rhs! function for the PlantHydraulicsModel.
 The rhs! function must comply with a rhs function of OrdinaryDiffEq.jl.
+
+Note that this rhs function updates the auxiliary variables as well. We chose to
+do so here because it reduces to a single loop over the layers, instead of looping
+over them in the auxiliary function as well. 
+
+Below, `fa` denotes a flux multiplied by the relevant cross section.
 """
 function make_rhs(model::PlantHydraulicsModel)
     function rhs!(dY, Y, p, t)
         @unpack vg_α, vg_n, vg_m, S_s, ν, K_sat, area_index = model.parameters
-        n_stem = model.domain.n_stem
-        n_leaf = model.domain.n_leaf
+        n_stem = model.n_stem
+        n_leaf = model.n_leaf
 
-        S_l = effective_saturation.(ν, Y.vegetation.ϑ_l)
-        ψ = water_retention_curve.(vg_α, vg_n, vg_m, S_l, ν, S_s)
-        for i in 1:(n_stem + n_leaf)
-            if i == 1
-                flux_in = flux_out_roots(model.root_extraction, model, Y, p, t)
-                area_index_in = area_index[:root]
-            else
-                flux_in = flux(
-                    model.domain.compartment_midpoints[i - 1],
-                    model.domain.compartment_midpoints[i],
-                    ψ[i - 1],
-                    ψ[i],
+        ψ = p.vegetation.ψ
+        ϑ_l = Y.vegetation.ϑ_l
+        fa = p.vegetation.fa
+
+        # boundary fluxes * cross sections
+        fa0 =
+            flux_out_roots(model.root_extraction, model, Y, p, t) .*
+            area_index[:root]
+        fa[n_stem + n_leaf] .=
+            transpiration(model.transpiration, t) .* area_index[:leaf]
+
+        # initialize first index
+        @. ψ[1] = water_retention_curve(
+            vg_α,
+            vg_n,
+            vg_m,
+            effective_saturation(ν, ϑ_l[1]),
+            ν,
+            S_s,
+        )
+
+        @inbounds for i in 1:(n_stem + n_leaf)
+            AIdz =
+                area_index[model.compartment_labels[i]] * (
+                    model.compartment_surfaces[i + 1] -
```

Comment:
would it be more natural to store the height of the compartments?

Reply:
I think we for sure will need canopy height as as parameter. I was also thinking that because we only ever need use the compartment surfaces to get the width of each layer, that maybe it would make sense to just pass the width of the layer (my mistake, I wasnt clear by referring to this as the height).

We can do this in a future PR if we want.